service/src/test/java/bio/terra/cbas/dao/TestRunSetDao.java

```diff
@@ -98,36 +100,87 @@ void retrievesAllRunSets() {
-    runSetDao.createRunSet(
+    RunSet laterRunSet =
         new RunSet(
-            laterRunSetId,
+            UUID.randomUUID(),
             methodVersion,
             "fetch_sra_to_bam workflow",
             "fetch_sra_to_bam sample submission",
             false,
             false,
             CbasRunSetStatus.COMPLETE,
-            OffsetDateTime.parse("2023-01-28T19:21:24.563932Z"),
-            OffsetDateTime.parse("2023-01-28T19:21:24.563932Z"),
-            OffsetDateTime.parse("2023-01-28T19:21:24.563932Z"),
+            runSet.submissionTimestamp().plusDays(1),
+            runSet.lastModifiedTimestamp().plusDays(1),
+            runSet.lastPolledTimestamp().plusDays(1),
             0,
             0,
             "[]",
             "[]",
             "sample",
             "user-foo",
-            workspaceId));
+            workspaceId);
+    runSetDao.createRunSet(laterRunSet);
+
+    RunSet latestRunSet =
+        new RunSet(
+            UUID.randomUUID(),
+            methodVersion,
+            "fetch_sra_to_bam workflow",
+            "fetch_sra_to_bam sample submission",
+            false,
+            false,
+            CbasRunSetStatus.COMPLETE,
+            runSet.submissionTimestamp().plusDays(2),
+            runSet.lastModifiedTimestamp().plusDays(2),
+            runSet.lastPolledTimestamp().plusDays(2),
+            0,
+            0,
+            "[]",
+            "[]",
+            "sample",
+            "user-foo",
+            workspaceId);
+
+    runSetDao.createRunSet(latestRunSet);
 
     assertEquals(
-        laterRunSetId, runSetDao.getLatestRunSetWithMethodId(method.methodId()).runSetId());
+        latestRunSet.runSetId(),
+        runSetDao.getRunSetsWithMethodId(method.methodId()).get(0).runSetId());
+    assertEquals(
+        laterRunSet.runSetId(),
+        runSetDao.getRunSetsWithMethodId(method.methodId()).get(1).runSetId());
+    assertEquals(
+        runSet.runSetId(), runSetDao.getRunSetsWithMethodId(method.methodId()).get(2).runSetId());
+
+    assertEquals(3, runSetDao.getRunSetsWithMethodId(method.methodId()).size());
+  }
+
+  @Test
+  void updateIsTemplate() {
+    runSetDao.updateIsTemplate(runSet.runSetId(), true);
+    RunSet updatedRunSetIsTemplate = runSetDao.getRunSet(runSet.runSetId());
+    assertTrue(updatedRunSetIsTemplate.isTemplate());
+
+    runSetDao.updateIsTemplate(updatedRunSetIsTemplate.runSetId(), false);
+    RunSet updatedRunSetIsNotTemplate = runSetDao.getRunSet(runSet.runSetId());
+    assertFalse(updatedRunSetIsNotTemplate.isTemplate());
+  }
+
+  @Test
+  void deleteRunSet() {
+    int response = runSetDao.deleteRunSet(runSet.runSetId());
+    assertEquals(1, response);
+
+    try {
+      RunSet deletedRunSet = runSetDao.getRunSet(runSet.runSetId());
+      Assertions.fail("Expected an exception, but none was thrown.");
+    } catch (Exception e) {
+      assertEquals("Index 0 out of bounds for length 0", e.getMessage());
+    }
```

Comment:
There's an `assertThrows` syntax used elsewhere in CBAS that might cover this case a little more neatly